### PR TITLE
Reduce singletons scope of impact

### PR DESCRIPTION
**Pull request by bitmouse**
Also remove desktop-dev-tools as unused component which is broken for a long time.

### CODE REVIEW

File: src/ipc/dbus/DBusManager.h [hunk 1]
```diff
@@ -46,34 +47,39 @@ class DBusObjectRegistry;
...
-class FaceliftIPCLibDBus_EXPORT DBusManager
+class FaceliftIPCLibDBus_EXPORT DBusManager : public DBusManagerInterface
 {
-
 public:
-    DBusManager();
+    DBusManager(const DBusManager&) = delete;
+    DBusManager(const DBusManager&&) = delete;
+    DBusManager& operator=(const DBusManager&) = delete;
+    DBusManager& operator=(const DBusManager&&) = delete;
```
[the remark this answers is not in the document]
Comment (bitmouse) [reply src/ipc/dbus/DBusManager.h:56]: =delete should stay in header file.

File: src/ipc/ipc-common/InterfaceManager.h [hunk 2]
```diff
@@ -41,40 +42,41 @@
...
-class FaceliftIPCCommonLib_EXPORT InterfaceManager : public QObject
+class FaceliftIPCCommonLib_EXPORT InterfaceManager : public InterfaceManagerInterface
 {
     Q_OBJECT
 
 public:
+    InterfaceManager(const InterfaceManager&) = delete;
+    InterfaceManager(const InterfaceManager&&) = delete;
+    InterfaceManager& operator=(const InterfaceManager&) = delete;
+    InterfaceManager& operator=(const InterfaceManager&&) = delete;
```
[the remark this answers is not in the document]
Comment (bitmouse) [reply src/ipc/ipc-common/InterfaceManager.h:58]: as above

File: src/ipc/ipc-common/InterfaceManagerInterface.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#endif
+
+namespace facelift {
+
+class NewIPCServiceAdapterBase;
+
+class FaceliftIPCCommonLib_EXPORT InterfaceManagerInterface : public QObject
+{
+    Q_OBJECT
+public:
```
[the remark this answers is not in the document]
Comment (bitmouse) [reply src/ipc/ipc-common/InterfaceManagerInterface.h:49]: No need. Class is derived from QObject which already provides virtual destructor.

File: src/model/InterfaceBase.cpp [hunk 4]
```diff
@@ -47,6 +47,7 @@ void InterfaceBase::setComponentCompleted() {
 
 void registerInterfaceImplementationInstance(InterfaceBase & i)
 {
+  Q_UNUSED(i);
```
[the remark this answers is not in the document]
Comment (bitmouse) [reply src/model/InterfaceBase.cpp:50]: removing argument name at all. No need to keep commented out code.

File: src/ipc/ipc-common/InterfaceManagerInterface.h [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    virtual void registerAdapter(const QString &objectPath, NewIPCServiceAdapterBase *adapter) = 0;
+    virtual void unregisterAdapter(NewIPCServiceAdapterBase *adapter) = 0;
+    virtual NewIPCServiceAdapterBase *getAdapter(const QString &objectPath) = 0;
+    virtual Registry<QPointer<NewIPCServiceAdapterBase>>& content() = 0;
```
[the remark this answers is not in the document]
Comment (bitmouse) [reply src/ipc/ipc-common/InterfaceManagerInterface.h:54]: Done

**Review by kunichik — APPROVED** [review]

**Pull request merged**